pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java

```diff
@@ -303,13 +302,12 @@ public void addFailed(ManagedLedgerException exception, Object ctx) {
         @Override
         public void run() {
             try {
-                while (lastConfirmedEntry.compareTo(currentLoadPosition) > 0) {
-                    if (((ManagedCursorImpl) cursor).isClosed()) {
-                        log.warn("[{}] MLPendingAckStore cursor have been closed, close replay thread.",
-                                cursor.getManagedLedger().getName());
-                        return;
-                    }
-                    fillEntryQueueCallback.fillQueue();
+                if (cursor.isClosed()) {
```

Comment:
why did you move this outside of the _while_ loop ?

Reply:
if cursor close will return ManagedledgerfenceException, so can delete this code, to speed up recovery